### DISCUSSION TITLE
Proper comparison for custom type with `date` column type.

### DISCUSSION THREAD

**Discussion by mjroeleveld** [mikro-orm/mikro-orm#5164] [Q&A]
When creating a custom type with a `date` column type (below), the property always is updated when the entity is retrieved, even though it's value didn't change.

```typescript
  getColumnType(prop: EntityProperty, platform: Platform): string {
    return platform.getDateTypeDeclarationSQL(prop.length);
  }
```

A `findOne` and a `flush` of an entity with a property of this type result in in update even though it's value didn't change:

```
1b[90m[query] \u001b[39mupdate \"test_entity\" set \"date_of_birth\" = '1990-1-1' where \"id\" = 1\u001b[90m [took 2 ms, 1 row affected]\u001b[39m
```

What should I do for the compare to work?

**Comment by B4nan** [comment]
You'll need to provide more details.

> 1b[90m[query] \u001b[39mupdate \"test_entity\" set \"date_of_birth\" = '1990-1-1' where \"id\" = 1\u001b[90m [took 2 ms, 1 row affected]\u001b[39m

The `1990-1-1` is not a valid date, sounds like your type is mapping things wrong? It should be `1990-01-01`.
